ls1intum/Artemis
# [Feature] Grading System - Display Grades and Grading Key to Students for Courses

### Checklist

### Motivation and Context


The students couldn't see their grades or bonus points for a course yet, in case a grading key was defined for that course. Now they would see their grades on the course statistics page after the course is done. They would also be able to view the grading key.

### Description

I refactored the grading key overview page, so it's a separate module and can be accessed both from courses and exams. I also added the grades on the course statistics page, below the overall points "statistics", and a button that leads to the grading key overview page.

### Steps for Testing


0. If you don't want to do prep for the testing, using my course (IvoTC) on TS3, with the test users 1, 2 (students) and your admin should suffice.
1. Log in to Artemis as an Instructor and set a "Grade" grading key for a course
2. Then, log in as a student of that course (who has participated in a few exercises) and go to the course statistics page
3. Check if the grade of the student is correct
4. Click on the View Grading Key button and check if the grading key is displayed correctly
5. Log back in as an instructor, and change the grading key to type "Bonus"
6. Log in again as a student, and check if the bonus points are displayed correctly -> also navigate to the grading key page and view if it's correct.
7. Finally, delete the grading key as an instructor -> log in as a student -> the grade information shouldn't be visible

### Test Coverage



GradeStepRepository.java: 100%
grading-system.service.ts: 90.74%
grading-key-overview.component.ts: 93.48%

### Screenshots


Course Statistics Page (student):
![Screenshot 2021-06-24 231657](https://user-images.githubusercontent.com/63168603/123326870-6c82c780-d542-11eb-8a6c-3977163f3845.png)
Grading Key Overview Page:
![Screenshot 2021-06-24 231718](https://user-images.githubusercontent.com/63168603/123326874-6d1b5e00-d542-11eb-9745-4c140bcc3bc0.png)
